acceptance/analyzer_test.go

```diff
@@ -1015,6 +1053,247 @@ func testAnalyzerFunc(platformAPI string) func(t *testing.T, when spec.G, it spe
 				assertAnalyzedMetadata(t, filepath.Join(copyDir, "some-other-layers", "analyzed.toml")) // analyzed.toml is written at the provided -layers directory: /some-other-layers
 			})
 		})
+
+		when("validating stack", func() {
+			it.Before(func() {
+				h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) < 0, "Platform API < 0.7 does not validate stack")
+			})
+
+			when("stack metadata is not present anywhere", func() {
+				it("fails validation", func() {
+					cmd := exec.Command(
+						"docker", "run", "--rm",
+						"--network", registryNetwork,
+						"--env", "CNB_PLATFORM_API="+platformAPI,
+						"--env", "CNB_STACK_PATH=/cnb/fake-stack.toml",
+						analyzeImage,
+						ctrPath(analyzerPath),
+					) // #nosec G204
+					output, err := cmd.CombinedOutput()
+
+					h.AssertNotNil(t, err)
+					expected := "failed to validate stack: failed to resolve stack"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+
+			when("stack metadata is present", func() {
+				when("stacks match", func() {
+					it("passes validation", func() {
+						h.SkipIf(t, runtime.GOOS == "windows", "Not yet working on Windows")
+
+						execArgs := []string{ctrPath(analyzerPath)}
+
+						h.DockerRunAndCopy(t,
+							containerName,
+							copyDir,
+							ctrPath("/layers/analyzed.toml"),
+							analyzeImage,
+							h.WithFlags(
+								"--network", registryNetwork,
+								"--env", "CNB_PLATFORM_API="+platformAPI,
+							),
+							h.WithArgs(execArgs...),
+						)
+					})
+				})
+
+				when("stacks do not match", func() {
```

Comment:
At first glance this seems like a lot of logic to be testing at the acceptance level. I wonder if we could pull the stack validation into a separate unit-testable service. I feel like our acceptance tests should really only be testing that the correct file is being read, args are respected, etc. But I haven't read through the whole PR yet so let me know if there is something I am missing here!

Reply:
Yeah - a lot of the validation in other areas is in the `cmd/`, but I agree it feels like too much. Both the `exporter` and `rebaser` do the run image resolution in the `cmd/`. Do you want me to take a go at moving things around?

Reply:
I took a swing at breaking things apart in 50dd6af611b70a935607791d9ebbf4e68822a4a4 and ec87c177d14e331536dc789acda5ba67ab6a690e. Let me know what you think. 

I find it hard to move things around in this repo. The line of what belongs in the cmd vs what belongs in the analyzer is blurry at best. We have things like `dockerClient` that we tend to not leak into the analyzer, for instance. But that seam makes it difficult to write something like a `ImageResolver` that has an interface like `resolver.Resolve(runImageRef)` that abstracts away the remote vs local. So that local vs remote logic ends up in each command. 

Maybe giving `lifecycle.Analyzer` an `imageResolver: defaultImageResolver(useDaemon, dockerClient, regCreds)` and push _all_ the image resolution inside could work out. Make the cmds as anemic as possible so that we can unit test. We could also have `fileResolver: defaultFileResolver()` that we use to read files that we can mock out as well to push all the reads into the analyzer and out of the cmd.

This feels like a larger discussion though...